app/services/mediator.rb

```diff
@@ -1,15 +1,28 @@
 module Mediator
   class DocumentHasNoFile < StandardError; end
+  class DocumentNotChunkable < StandardError; end
+
+  # Returns true if the document's content type is chunkable
+  def self.chunkable?(document)
+    document.file.attached? && Parsers.textual?(document.content_type)
+  end
+
+  # Return true if document needs to be fetched?
+  def self.fetchable?(document)
+    document.filename.nil?
```

Comment:
This doesn't necessarily make the Document fetchable. It's more that it requires fetching, or needs to be fetched, or doesn't have a file.

Reply:
I didn't like this either; new impl coming with something more expressive

Reply:
Done.